c10/core/DispatchKey.h

```diff
@@ -240,6 +242,10 @@ enum class DispatchKey : uint16_t {
 
   ZeroTensor, // registered at build/aten/src/ATen/RegisterZeroTensor.cpp
 
+  // The CsrTranspose dispatch key is set for any SparseCsr tensor
+  // that is a sparse CSC tensor.
+  CsrTranspose,
```

Comment:
We could also exactly copy the SparseCsrCPU, SparseCsrCUDA setup by introducing SparseCscCPU and SparseCscCUDA keys and then [use those during construction](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/sparse/SparseCsrTensor.cpp#L141-L161).

Reply:
Indeed. It basically means that sparse CSC will be independent of sparse CSR format, but nevertheless, the existing CSR implementations can be reused to implement CSC features.

Reply:
@pearu - yes, but we can achieve that by generalizing them and adding them as kernels to various native functions. At least this way we can track coverage and prevent bugs (accidentally calling a CSR function with a CSC Tensor). I'd then also employ the same strategy for BSR and BSC.

I suggest we go with adding SparseCscCPU and SparseCscCUDA and then sending another PR to unify our logic around dispatch keys + layouts.

Reply:
@pearu Ok, after further offline discussion, can we do this without introducing new dispatch keys at all?

Reply:
Wrote up an issue around that https://github.com/pytorch/pytorch/issues/74916